python_modules/libraries/dagster-aws/dagster_aws_tests/pipes_tests/test_pipes.py

```diff
@@ -935,6 +937,45 @@ def materialize_asset(env, return_dict):
         assert return_dict[0]["tasks"][0]["stoppedReason"] == "Dagster process was interrupted"
 
 
+def test_ecs_pipes_waiter_config(pipes_ecs_client: PipesECSClient):
+    with instance_for_test() as instance:
+        """
+        Test Exception is thrown when the wait delay is less than the processing time.
+        """
+        os.environ.update({"WAIT_DELAY": "1", "WAIT_MAX_ATTEMPTS": "1", "SLEEP_SECONDS": "2"})
+        try:
+            materialize([ecs_asset], instance=instance, resources={"pipes_ecs_client": pipes_ecs_client})
+            assert False
+        except Exception:
+            assert True
+
+        """
+        Test Exception is thrown when the wait attempts * wait delay is less than the processing time.
+        """
+        os.environ.update({"WAIT_DELAY": "1", "WAIT_MAX_ATTEMPTS": "2", "SLEEP_SECONDS": "3"})
+        try:
+            materialize([ecs_asset], instance=instance, resources={"pipes_ecs_client": pipes_ecs_client})
+            assert False
+        except Exception:
+            assert True
+
+        """
+        Test asset is materialized successfully when the wait delay is greater than the processing time.
+        """
+        os.environ.update({"WAIT_DELAY": "10", "WAIT_MAX_ATTEMPTS": "1", "SLEEP_SECONDS": "2"})
+        materialize([ecs_asset], instance=instance, resources={"pipes_ecs_client": pipes_ecs_client})
+        mat = instance.get_latest_materialization_event(ecs_asset.key)
+        assert mat and mat.asset_materialization
+
+        """
+        Test asset is materialized successfully when the wait attempts * wait delay is greater than the processing time.
+        """
+        os.environ.update({"WAIT_DELAY": "1", "WAIT_MAX_ATTEMPTS": "10", "SLEEP_SECONDS": "2"})
+        materialize([ecs_asset], instance=instance, resources={"pipes_ecs_client": pipes_ecs_client})
+        mat = instance.get_latest_materialization_event(ecs_asset.key)
+        assert mat and mat.asset_materialization
```

Comment:
Obviously sleeping this much will slow the test suite considerably. Let me know if you'd rather consolidate it down to just two tests (1 happy path, 1 non happy path) to reduce this a bit. Unfortunately, the wait delay has to be an seconds integer so can't use sub-second sleeps.